client/src/client.rs

```diff
@@ -22,9 +23,80 @@ use small::SmallStr;
 
 use crate::{
     config::Configuration,
-    http_client::{self, StatusCode, WebSocketError, WebSocketMessage},
+    http::{Headers as HttpHeaders, Method as HttpMethod, RequestBuilder, Response, StatusCode},
+    http_default::{self, DefaultRequestBuilder, WebSocketError, WebSocketMessage},
 };
 
+/// General trait for all response handlers
+pub trait ResponseHandler<O, T = Vec<u8>> {
```

Comment:
Personally I think you should consider making one an associated type.

Reply:
Done